analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd

```diff
@@ -0,0 +1,263 @@
+}
+```
+
+```{r warning=FALSE}
+library(tidyverse)
+```
+
+### Directory and files
+
+#### Directories
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+results_dir <- "results"
+```
+
+#### Input
+
+When we run this locally, we want to tie it to a specific version of the histologies file _prior_ to feedback we received from pathology that is recorded on [#609](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609).
+
+```{r}
+if (running_in_ci) {
+  histologies_file <- file.path(data_dir, "pbta-histologies.tsv")
+} else {
+  histologies_file <- file.path(data_dir, "release-v15-20200228",
+                                "pbta-histologies.tsv")
+}
+```
+
+```{r}
+compiled_results_file <- file.path(results_dir, 
+                                   "compiled_molecular_subtypes.tsv")
+fusions_file <- file.path(data_dir,
+                          "pbta-fusion-putative-oncogenic.tsv")
+```
+
+#### Output
+
+```{r}
+output_file <- file.path(results_dir,
+                         "compiled_molecular_subtypes_with_pathology_feedback.tsv")
+```
+
+## Read in data
+
+```{r message=FALSE}
+histologies_df <- read_tsv(histologies_file, guess_max = 10000)
+compiled_df <- read_tsv(compiled_results_file)
+fusions_df <- read_tsv(fusions_file)
+```
+
+## Revise subtyping calls based on pathology feedback
+
+### `PT_7E3V3JFX`
+
+`PT_7E3V3JFX` was classified in `molecular-subtyping-HGG` as `DMG, H3 K28` because the first step in that module was to identify any samples in the cohort with "defining lesions:" H3 K28 or H3 G35 (original subtyping issue: [#249](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249)).
+
+```{r}
+compiled_df %>%
+  filter(Kids_First_Participant_ID == "PT_7E3V3JFX")
+```
+
+The relevant notes above are
+
+> 1. `PT_7E3V3JFX` specimens were consistent with the original EPN dx, so pathology would call this a rare EPN, H3 K28 mutated tumor, rather than DMG.
+> 3. Because 1 is a rare tumor (maybe first seen), the logic of searching for all H3 K28 mutations in [HGG subtyping](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249) would convert this sample by default - how to handle this? 
+
+So we will revise the `molecular_subtype`, `integrated_diagnosis`, `short_histology`, and `broad_histology` accordingly.
+
+```{r}
+compiled_df <- compiled_df %>%
+  mutate(
+    integrated_diagnosis = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "Ependymoma",
+      TRUE ~ integrated_diagnosis 
+    ),
+    short_histology = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "Ependymoma",
+      TRUE ~ short_histology
+    ),
+    broad_histology = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "Ependymal tumor",
+      TRUE ~ broad_histology
+    ),
+    molecular_subtype = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "EPN, H3 K28",
+      TRUE ~ molecular_subtype
+    )
+  )
+```
```

Comment:
Like you've checked for BRAF fusion in the newly changed HGG => LGG,BRAF V600E subtype below,  do you think we need to check for any other subtyping ( I think they might be subtyped as RELA/YAP1 fusion ) in ependymoma along with the EPN ,H3 K28M ?

Reply:
The ependymoma table hasn't been completed yet (see open PR #555), so there's no EPN table included in `01-compile-subtyping-results.Rmd`. So I think it makes sense to tackle this suggestion at the same time as adding EPN into `molecular-subtyping-pathology` more generally.

Reply:
Once #645 gets closed, I will file an update an analysis ticket about all the EPN todos.

Reply:
ok sounds good!